fix: support ordering by key for multi queries

Primevenn opened the pull request:
fix: support ordering by key for multi queries

Fixes #629

Primevenn committed:
fix: support ordering by key for multi queries (#630)

Fixes: #629

Primevenn commented:
@googlebot I signed it!

Commit:
Merge branch 'master' into fix/multiquery_order_by_key

chrisrossi commented on google/cloud/ndb/_datastore_query.py:
```diff
@@ -801,10 +801,22 @@ def _compare(self, other):
-            other_value = helpers._get_value_from_value_pb(other_value_pb)
+
+            if order.name == '__key__':
+                this_value = helpers.key_from_protobuf(self.result_pb.entity.key)
```
I think you could probably just append `.flat_path` to the ends of these two lines and save the extra `if` statements below.

chrisrossi commented on google/cloud/ndb/_datastore_query.py:
```diff
@@ -801,10 +801,22 @@ def _compare(self, other):
+                other_value_pb = other.result_pb.entity.properties[order.name]
+                other_value = helpers._get_value_from_value_pb(other_value_pb)
+
+            # Compare key paths if ordering by key
```
See above.

chrisrossi commented:
@Primevenn This PR looks great. Thank you for submitting it. 

In order to accept this PR we'll need unit test coverage for the new code. There are also some linting errors that can be cleaned up by running `black`. (`nox -e blacken`)

Once you can run `nox` locally and not get any errors, you should be about good to go.

Primevenn committed:
fix: Add unit test coverage

Primevenn commented:
@chrisrossi Thanks for the review!

As u suggested i've appended flat_path to the 2 lines  under the if statement that handles ordering Entity __key__and have
moved the other if statements inside the else block. 

These 2 if statements are necessary for ordering by keyProperty since `helpers._get_value_from_value_pb()` will return Key objects for `key_value` fields.

I've also added unit test coverage for these 2 scenarios and ran `black`

chrisrossi reviewed (approved):
Excellent. Thank you!

Commit:
Merge branch 'master' into fix/multiquery_order_by_key
